api/v1beta1/virtinkcluster_types.go

```diff
@@ -17,15 +17,33 @@ type VirtinkClusterSpec struct {
 	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
 	ControlPlaneEndpoint capiv1beta1.APIEndpoint `json:"controlPlaneEndpoint,omitempty"`
 
-	// ControlPlaneServiceType can be used to modify type of service that fronts the control plane nodes to handle the
+	// ControlPlaneServiceTemplate can be used to modify service that fronts the control plane nodes to handle the
 	// api-server traffic (port 6443). This field is optional, by default control plane nodes will use a service
-	// of type ClusterIP, which will make workload cluster only accessible within the same cluster.
-	ControlPlaneServiceType *corev1.ServiceType `json:"controlPlaneServiceType,omitempty"`
+	// of type ClusterIP, which will make workload cluster only accessible within the same cluster. Note, this does
+	// not aim to expose the entire Service spec to users, but only provides capability to modify the service metadata
+	// and the service type. If the name and namespace are both set, the existed control plane service will be used,
+	// we only modify its selector to match the labels of control plane nodes, and this service will be preserved
+	// when Cluster is cleaned up. Otherwise a new control plane service will be created, and this new service
+	// will be cleaned up along with Cluster.
+	ControlPlaneServiceTemplate ControlPlaneServiceTemplate `json:"controlPlaneServiceTemplate,omitempty"`
```

Comment:
So we are creating a new CP service using this template, not using a pre-existent CP service?

Reply:
If both the name and namespace are set, the pre-existent CP Service will be used.

Reply:
That could be too vague to users. We may need a more explicit way

Reply:
Done.

Reply:
That seems reasonable. But the CP Services with additional annotations or labels have to be created manually, is this acceptable?